From d18cf76d73df8f9cc751d4b4ba5a635c70c0c645 Mon Sep 17 00:00:00 2001
From: Rich Felker <dalias@aerifal.cx>
Date: Sun, 30 Aug 2015 18:44:58 +0000
Subject: [PATCH] remove use of buggy .SECONDARY special target in makefile

this functionality is affected by GNU make bug #30653, "intermediate
files incorrectly pruned in parallel builds". on affected versions of
make, parallel builds attempt to compile source files before
alltypes.h is generated.

as noted with commit a91ebdcfac6804714a1fe39f4375e2b4ebab085b, which
added the use of .SECONDARY, suppression of removal of "intermediate"
files does not seem to be needed at present. if it is needed in the
future, it should be achievable by explicitly mentioning their names
as targets or prerequisites.
---
 Makefile | 2 --
 1 file changed, 2 deletions(-)

diff --git a/Makefile b/Makefile
index 07336d2..5a6a43b 100644
--- a/Makefile
+++ b/Makefile
@@ -203,6 +203,4 @@ musl-git-%.tar.gz: .git
 musl-%.tar.gz: .git
 	 git archive --format=tar.gz --prefix=$(patsubst %.tar.gz,%,$@)/ -o $@ v$(patsubst musl-%.tar.gz,%,$@)
 
-.SECONDARY:
-
 .PHONY: all clean install install-libs install-headers install-tools
-- 
2.5.1

